bundles/org.openhab.core.model.script/src.moved/test/java/org/openhab/core/model/script/actions/SemanticsTest.java

```diff
@@ -98,6 +104,21 @@ public void setup() throws ItemNotFoundException {
         equipmentItem.addMember(subEquipmentItem);
         subEquipmentItem.addGroupName(equipmentItem.getName());
 
+        SemanticTags.add("Indoor", Location.class);
```

Comment:
Relying on mutable `SemanticTags` singleton still leaves whole implementation vulnerable to changes performed via other channel. Access to this singleton does not seem to be guarded in any way.

Reply:
I understand your general point, even if at this place it is just to build the semantic model for a unit test.

I should be able to remove the static method `add` and rather do the job in the tag registry itself.
But then I will miss the method for unit tests.
I have to think about that but I agree we should avoid the static method to add a new tag class.

Reply:
If I correctly remember, I can solve the problem in unit tests by making the method protected.
To be checked later today.

Reply:
The unit tests do no more use `SemanticTags.add` to build the semantic tags but use the tag registry.
As a consequence, I should be able now to remove the static public method.

Reply:
This is now done, the static methods in `SemanticTags` are removed and only the tag registry can now create the tag classes at runtime. No static method is now available.